CasperLabs/casper-node
# NDRS-536: Validating blocks in API client

NDRS-536 - Client should validate block returned by get-block against block hash
https://casperlabs.atlassian.net/browse/NDRS-536

# Steps to QA Proper Functionality

The following commands should all run successfully without outputting anything to standard error.

1. Run `./run-dev.sh`
2. Check the nodes are up and listening on TCP ports with `lsof -i -P -n | grep LISTEN`. There should be 10 `casper-no` ports, and one of them should be 7777
3. Get the latest block hash with

```bash
# LATEST_HASH=$(cargo run --quiet -p casper-client -- get-block -n http://localhost:7777 | jq '.result.block.hash' | sed -e 's/"//g')
# echo $?
```

...the result of `echo $?` should be `0` indicating success.

4. Use the client to get the `$LATEST_HASH`, run:

```bash
# cargo run --quiet -p casper-client -- get-block -n http://localhost:7777 -b $LATEST_BLOCK
```

This should output something like:

```json
{
  "jsonrpc": "2.0",
  "result": {
    "api_version": "1.0.0",
    "block": {
      "body": null,
      "hash": "9ae64e40de59f467baf50605be07b5d7ef85055be15a8e24aeddae6a77fd683a",
      "header": {
        "accumulated_seed": "9b3b55568d333e8bd3760daab413108ede63d4ff0ec8b3fc41fc307babffc6df",
        "body_hash": "0e5751c026e543b2e8ab2eb06099daa1d1e5df47778f7787faab45cdf12fe3a8",
        "deploy_hashes": [],
        "era_end": null,
        "era_id": 354,
        "height": 3896,
        "parent_hash": "dd554d258a8c88fb607664c364a7efb407ddddbce003ccf1c67c4470c4064b39",
        "proposer": "018f5a3ee4c1221686fcdbe6c0b6168acb24025c5485f59f7c4039ffc444fb7509",
        "random_bit": true,
        "state_root_hash": "da41fd101286aec5e37368ed1736d3e4e6c62f2eea8a9d7bce192904b0bfc086",
        "timestamp": "2020-10-29T13:33:56.608Z"
      },
      "proofs": [
        "01e49b83c62131c65d47233cc2643cfceb1e472d7c20c7f143a91d024b614627abbb4d2b5af88743fb678f0df1211e1694d99f6149f74e200cde4067ded033ff03"
      ]
    }
  },
  "id": 1780427375
}
```
